File: padroesCriacionais/padrao-de-projeto-factory/java/src/main/java/factory/PagarMe.java
package factory;

public class PagarMe implements iIntermediadorPagamento {

    @Override
    public String pix(String dadosPedido, String dadosUsuario) {
        return "dados do pagamento com pix na Pagarme";
    }

    @Override
    public String boleto(String dadosPedido, String dadosUsuario) {
        return "dados do pagamento com boleto na Pagarme";
    }

    @Override
    public String credito(String dadosPedido, String dadosUsuario) {
        return "dados do pagamento com credito na Pagarme";
    }
}

File: padroesCriacionais/padrao-de-projeto-factory/php/src/PagarMe.php
<?php

namespace Factory\Php;

class PagarMe implements iIntermediadorPagamento
{
    public function pix(string $dadosPedido, string $dadosUsuario): String
    {
        return "dados do pagamento com pix na Pagarme";
    }

    public function boleto(string $dadosPedido, string $dadosUsuario): String
    {
        return "dados do pagamento com boleto na Pagarme";
    }

    public function credito(string $dadosPedido, string $dadosUsuario): String
    {
        return "dados do pagamento com credito na Pagarme";
    }
}